### ISSUE TITLE
Category Search Broken

### ISSUE THREAD

**Issue by enrmbigg** [bjorn2404/jQuery-Store-Locator-Plugin#93]
When searching with categories if you search first an select a few categories, upon unchecking the search filter the results are limited to those previously whittled down.

**Comment by enrmbigg** [comment]
Example

    London: restaurants
    Paris: restaurants 
    Berlin

restaurants ticked shows

    London: restaurants
    Paris: restaurants 

Then after the search with restaurant unchecked shows, only those previous two when Berlin should now be available again

    London: restaurants
    Paris: restaurants

**Comment by bjorn2404** [comment]
I'm not sure I follow. Do you have an example link?

**Comment by enrmbigg** [comment]
Sorry currently I don't am afraid, am terrible at explain sometimes aha,

Let me try again.

So after doing a search with some categories checked I expect the full list of retailers to be available upon the next search but it seems that it takes the results from the previous search and uses that and not the original list

**Comment by bjorn2404** [comment]
With the categories still checked? It works that way so that options can be selected before the initial search.

**Comment by enrmbigg** [comment]
it seems to error on line 1885 saying "Cannot read property 'distance' of undefined" after the intial search

**Comment by bjorn2404** [comment]
Can you send an example link? You can email it to bjorn2404@gmail.com if you'd rather not post it here.

**Comment by enrmbigg** [comment]
Hi Bjorn,

Sorry its been a while,

So it looks like using a checkbox for the categories is a little funky.
If you do a search with i few categories selected, then untick and search it doesn't clear the filters in the js it seems

![example error](https://cloud.githubusercontent.com/assets/2420325/15607318/c6a93e8c-2409-11e6-9804-b78ed5911d55.png)

**Comment by enrmbigg** [comment]
Figured it out, bh-sl-filters need to exist on the filters otherwise you fail to un select filters
